### PR TITLE
[[ Message Box ]] Search script of current card as well as stack for send completion

### LINKED ISSUE #705: Msg box displays execution errors instead of parse errors
screen shot 2015-12-03 at 11 23 08

### PR DESCRIPTION
**Pull request by livecodeali**
Closes #705
Both the script of the intelligence object and the script of the
current card and stack should be searched for possible completion
of commands in the script editor.

To achieve this I've refactored the logic that works out if the message 
makes sense in the context of a given object, and made sure to try it
in the context of the current card if it doesn't make sense in the context 
of the intelligence object.

### CODE REVIEW

File: Toolset/libraries/revidelibrary.8.livecodescript [hunk 1]
```diff
@@ -9463,63 +9499,25 @@ command ideExecuteScript pScript, pDefaultStack, pIntelligenceObject, pDebugMode
    end if
    
    -- check if word 1 of pScript is a handler in the intelligence object's script or the default stack's script
-   local tScript, tScriptLine, tScriptError
+   local tIntellgenceScript, tCardStackScript, tScriptLine, tScriptError
```
Comment (peter-b) [inline Toolset/libraries/revidelibrary.8.livecodescript]: Missing "i" in "Intelligence"

Comment (livecodeali) [inline Toolset/libraries/revidelibrary.8.livecodescript]: Sounds like an insult somehow :-)

**Comment by livecodeali** [comment]
@peter-b Updated, thanks.

**Comment by runrevelanor** [comment]
@livecode-vulcan review ok  20cb4e5

**Comment by livecode-vulcan** [comment]
💙 review by @runrevelanor ok 20cb4e506c911681adb06895941886d9478bb208

**Pull request merged**